src/frontend/src/catalog/system_catalog/rw_catalog/iceberg_tables.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::{anyhow, Context};
+use risingwave_batch::executor::postgres_row_to_owned_row;
+use risingwave_common::acl::AclMode;
+use risingwave_common::bail;
+use risingwave_common::catalog::{Field, Schema};
+use risingwave_common::types::DataType::Varchar;
+use risingwave_common::types::Fields;
+use risingwave_frontend_macro::system_catalog;
+use risingwave_pb::user::grant_privilege::Object as GrantObject;
+use thiserror_ext::AsReport;
+
+use crate::catalog::root_catalog::SchemaPath;
+use crate::catalog::system_catalog::SysCatalogReaderImpl;
+use crate::error::Result;
+
+#[derive(Fields)]
+#[primary_key(catalog_name, table_namespace, table_name)]
+struct IcebergTables {
+    pub catalog_name: String,
+    pub table_namespace: String,
+    pub table_name: String,
+    pub metadata_location: Option<String>,
+    pub previous_metadata_location: Option<String>,
+    pub iceberg_type: Option<String>,
+}
+
+#[system_catalog(table, "rw_catalog.iceberg_tables")]
+async fn read(reader: &SysCatalogReaderImpl) -> Result<Vec<IcebergTables>> {
+    let Ok(meta_store_endpoint) = std::env::var("RW_SQL_ENDPOINT") else {
```

Comment:
Note that these env vars `RW_SQL_...` are all applied to the Meta node, but here it's in the frontend node. Does this mean the Nimtable instances are always deployed as standalone node? Or perhaps other similar assumptions that I don't know?

Anyway It's acceptable to me because we want a quick & dirty solution right now, but it's worth noting down some comments as a reminder.

Reply:
For nimtable, we need `RW_SQL...` and `NIMTABLE_S3...` to access iceberg table, so these envs are expected to be accessible in the frontend, compute, and meta nodes. And the standalone deployment mode is preferred because it makes the envs convenient to access.